### ISSUE TITLE
Investigate flaky logger tests

### ISSUE THREAD

**Issue by pared** [iterative/dvc#1873]
Sometimes, upon running parallel tests, test_logger fails.
Chance for failing seems to increase upon running 
`python -m tests -n={X}`
where X > num_cpu
and inserting sleep for 1 second before assertion in one of the test_logger.py tests.

Running with X < num_cpu seems to decrease chance of failing

```
self = <tests.unit.test_logger.TestColorFormatter object at 0x7efcf95d3c88>, caplog = <_pytest.logging.LogCaptureFixture object at 0x7efcf95f0908>

    def test_exception(self, caplog):
        try:
            raise ValueError
        except Exception:
            logger.exception("message")
    
        expected = (
            "{red}ERROR{nc}: message\n"
            "\n"
            "{footer}".format(footer=formatter.footer, **colors)
        )
    
        with caplog.at_level(logging.INFO, logger="dvc"):
>           assert expected == formatter.format(caplog.records[0])
E           IndexError: list index out of range

tests/unit/test_logger.py:70: IndexError

```

**Comment by pared** [comment]
might be usefult to reproduce:
```
#! /bin/bash
for i in {1..10}
do
  python -m tests -n=24 >> test_result.log
done

```

**Comment by mroutis** [comment]
@pared , this is interesting, I think it is because it generates different logs and `caplog` registers them, I will thinker with it and see if I can come up with a solution.

**Comment by pared** [comment]
To be honest I just had suspicion that we have some kind race condition and assumed that there will be higher chance of "test overlapping" if one is too slow. Tried it and it seems to work.

**Comment by mroutis** [comment]
Relevant: https://docs.pytest.org/en/latest/flaky.html

**Comment by efiop** [comment]
Fixed by #1875
